### DISCUSSION TITLE
Use componentFirst programmatically

### DISCUSSION THREAD

**Discussion by MallauryGerard** [GrapesJS/grapesjs#5033] [Q&A]
Hello,

I would like to create a button to enable or disable "componentFirst". So users will be able to choose if a style should be applied to all elements or only one element.

But it looks like styles applied with "componentFirst: true" take over the styles with componentFirst set to false.

How to reproduce : 
1. I set componentFirst to true
2. I change the background-color of a button (red)
3. I switch componentFirst to false
4. Then I change the background-color of the same button again (green)

The green color do not applies. The button is still red.

Do you know a solution?

I can not use "!important" because I already concat "!important" after all styles to be sure we can override bootstrap classes of my blocks...

Thanks!

**Comment by artf** [comment]
Yeah, the componentFirst rules have higher specificity than classes and it should work in that way. You can always add your custom functionality and swap styles on component rules
